lib/customer-deposit-wallet/test/unit/Cardano/Wallet/Deposit/HTTP/JSON/JSONSpec.hs

```diff
@@ -26,58 +32,117 @@ import Cardano.Wallet.Deposit.Read
 import Data.Aeson
     ( FromJSON (..)
     , ToJSON (..)
+    , Value
     , decode
     , encode
     )
+import Data.Aeson.Encode.Pretty
+    ( encodePretty
+    )
+import Data.OpenApi
+    ( Definitions
+    , Schema
+    , validateJSON
+    )
 import Test.Hspec
-    ( Spec
+    ( Expectation
+    , Spec
     , describe
     , it
+    , shouldBe
     )
 import Test.QuickCheck
     ( Arbitrary (..)
+    , Gen
     , Property
+    , Testable
     , arbitrarySizedBoundedIntegral
     , chooseInt
+    , counterexample
+    , forAll
     , property
     , shrinkIntegral
     , vectorOf
     , (===)
     )
 
 import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy.Char8 as BL
+import qualified Data.List as L
 
 spec :: Spec
-spec =
+spec = do
     describe "JSON serialization & deserialization" $ do
         it "ApiT Address" $ property $
             prop_jsonRoundtrip @(ApiT Address)
         it "ApiT Customer" $ property $
             prop_jsonRoundtrip @(ApiT Customer)
         it "ApiT CustomerList" $ property $
             prop_jsonRoundtrip @(ApiT CustomerList)
+    describe "schema checks" $ do
+        it "ApiT Address"
+            $ forAll genApiAddress
+            $ counterExampleJSON "validate"
+            $ validateInstance depositDefinitions addressSchema
```

Comment:
Since we have to repeat this for every type, I would suggest shrinking it down to a single line, e.g.

```suggestion
            $ jsonMatchesSchema genApiAddress addressSchema
```

where

```hs
jsonMatchesSchema :: (ToJSON a, FromJSON a) => Gen a → Schema → Property
jsonMatchesSchema gen schema =
    forAll gen
        $ counterExampleJSON "validate"
        $ validateInstance depositDefinitions schema
```